Tactics/Common.lean

```diff
@@ -78,3 +78,99 @@ def getStateFieldString? (e : Expr) : MetaM (Option String) := OptionT.run do
   | StateField.FLAG pExpr => getPFlagString? pExpr
   | StateField.ERR        => return "err"
   | _                     => panic! s!"[getStateFieldName?] Unexpected expression: {e}"
+
+/-! ## Reflection of literals (possibly after reduction) -/
+
+#check BitVec.ofFin
+
+/-- A wrapper around `Lean.Meta.getBitVecValue?`
+that additionally recognizes:
+- a `BitVec.ofFin (Fin.mk _ _)` application (which is the raw normal form)
+-/
+-- TODO: should this be upstreamed to core?
+def getBitVecValue? (e : Expr) : MetaM (Option ((n : Nat) × BitVec n)) :=
+  match_expr e with
+    | BitVec.ofFin _ i => OptionT.run do
+      let ⟨n, i⟩ ← getFinValue? i
+      let n' := Nat.log2 n
+      if h : n = 2^n' then
+        return ⟨n', .ofFin (Fin.cast h i)⟩
+      else
+        failure
```

Comment:
```suggestion
        throwError s!"Expected (BitVec.ofFin (Fin.mk {n} {i})) to have {n} be a power of two."
```

Reply:
Note that this is in an `OptionT`, so failure here will return `MetaM.pure none`, this function is not supposed to throw an error if it fails reflection